csvy/writers.py

```diff
@@ -95,10 +101,22 @@ def close(self) -> None:
     def writerow(self, row: Iterable[Any]) -> None:
         """Write a single row of data to the CSV file."""
         self._writer.writerow(row)
+        if self._flush_immediately:
+            self._file.flush()
 
     def writerows(self, rows: Iterable[Iterable[Any]]) -> None:
         """Write multiple rows of data to the CSV file."""
         self._writer.writerows(rows)
+        if self._flush_immediately:
+            self._file.flush()
+
+    @property
+    def flush_immediately(self) -> bool:
+        return self._flush_immediately
+
+    @flush_immediately.setter
+    def flush_immediately(self, flag: bool) -> None:
+        self._flush_immediately = flag
```

Comment:
I would not include any setter/getter for this. I wouldn't expect users to change the flush flag midway through the process. If we do want to allow that, just remove the underscore from `self._flush_immediately` and let them change it directly.